scm-core/src/main/java/sonia/scm/repository/RepositoryManager.java

```diff
@@ -96,4 +97,20 @@
    * @return all namespaces
    */
   Collection<String> getAllNamespaces();
+
+
+  /**
+   * Creates a new repository and afterwards executes the logic from the {@param afterCreation}.
+   *
+   * @param repository    the repository {@link Repository}
+   * @param afterCreation consumer which expects a repository {@link Repository}
+   * @return {@link Repository} the created repository
```

Comment:
Adding links to the param and return does make only noise and does not help, because the types are already listed in the javadoc.

```suggestion
   * @param repository the repository to create
   * @param afterCreation consumer which is executed after the repository was created
   * @return created repository
```